src/v/kafka/server/handlers/topics/topic_utils.h

```diff
@@ -45,7 +45,10 @@ template<typename T>
 requires TopicRequestItem<T> creatable_topic_result
 generate_error(T item, error_code code, const ss::sstring& msg) {
     return creatable_topic_result{
-      .name = item.name, .error_code = code, .error_message = msg};
+      .name = item.name,
+      .error_code = code,
+      .error_message = msg,
+      .topic_config_error_code = code};
```

Comment:
Compatibility check: this field is present in all versions, we just never set it to anything non-zero before. Is this correct?

Reply:
Yes and this is fine because in the generated code there will be an if statement that will omit writing this field if the version is < 5. 

So why did I add the version conditionals in `create_topics.cc`? Because even though the same applies there, we would be adding some overhead to do some work that would eventually be thrown out ( going to query topic configs if version < 5 only to later not encode the data over the wire )